src/components/typeAheadCheckboxes/__tests__/typeaheadCheckboxes.test.tsx

```diff
@@ -10,4 +12,88 @@ describe('TypeaheadCheckboxes', () => {
     const component = await shallowComponent(<TypeaheadCheckboxes {...props} />);
     expect(component).toMatchSnapshot('basic');
   });
+
+  it('should filter options, select one, and close dropdown when maxSelections is 1', async () => {
+    const onChange = jest.fn();
+
+    render(
+      <TypeaheadCheckboxes
+        options={[
+          { value: 'alpha', label: 'Alpha' },
+          { value: 'beta', label: 'Beta' }
+        ]}
+        selectedOptions={[]}
+        onChange={onChange}
+        maxSelections={1}
+      />
+    );
+
+    const input = screen.getByRole('combobox');
+    await userEvent.type(input, 'alp');
+
+    const alphaOption = await screen.findByText('Alpha');
+    await userEvent.click(alphaOption);
+
+    expect(onChange).toHaveBeenCalledWith(['alpha']);
+
+    // Wait for the menu to disappear
+    await waitFor(() => expect(screen.queryByRole('menu')).not.toBeInTheDocument());
+  });
+
+  it('should filter options, select multiple, and call onChange twice (controlled)', async () => {
+    const user = userEvent.setup();
+    const options = [
+      { value: 'alpha', label: 'Alpha' },
+      { value: 'beta', label: 'Beta' },
+      { value: 'gamma', label: 'Gamma' }
+    ];
+
+    /**
+     *
+     */
+    const Wrapper = () => {
```

Comment:
This test verifies that the TypeaheadCheckboxes component works correctly in controlled mode when allowing multiple selections. In the test, I wrap the component in a Wrapper that manages the selected options using React state. We simulate a user typing to filter the options, selecting "Alpha", then re-opening the dropdown, filtering, and selecting "Beta". After both selections, we check that the placeholder updates to "2 items selected", confirming that both options are selected. Finally, we assert that the dropdown menu closes automatically when the maximum number of selections (maxSelections=2) is reached. This test ensures that the component properly supports controlled selection, updates the UI, and enforces the selection limit as expected.

Reply:
ahhhh, I see now ! I downloaded the video you shared, I will open a slack thread to address it.